src/scheduler/Scheduler.cpp

```diff
@@ -137,14 +142,29 @@ void Scheduler::notifyExecutorShutdown(Executor* exec,
 
     std::string funcStr = faabric::util::funcToString(msg, false);
 
-    // Remove from warm executors
-    std::string execId = exec->id;
+    // Find in list of executors
+    int execIdx = -1;
     std::vector<std::shared_ptr<Executor>>& thisExecutors = executors[funcStr];
-    std::remove_if(thisExecutors.begin(),
-                   thisExecutors.end(),
-                   [&execId](const auto& p) { return p->id == execId; });
+    for (int i = 0; i < thisExecutors.size(); i++) {
+        if (thisExecutors.at(i)->id == exec->id) {
+            execIdx = i;
+            break;
+        }
+    }
```

Comment:
Ditto re. trying to use the stdlib for this, it came out really ugly and I think this is more readable (and editable).